oeps/architectural-decisions/oep-0064-arch-mobile-application-codebase-modernization.rst

```diff
@@ -0,0 +1,99 @@
+
+A robust mobile experience makes learning opportunities more accessible whether that is for learners who don't have ready access to personal computers, or need to fit learning into their schedules around work and family commitments.
+
+
+Specification 
+************* 
+
+The existing mobile repositories are maintained by edX / 2U and can be accessed here:
+
+* https://github.com/openedx/edx-app-ios
+* https://github.com/openedx/edx-app-android
+
+* Support for the existing mobile applications will continue through the Redwood release. Starting with the Quince release, we will designate the new repositories as the default Open edX mobile applications to encourage adoption and use of the new applications for community team members.
+* Once available, we will link to documentation within the new repositories to describe transition options for community members migrating from existing builds of the older codebases.
+* After the publication of the Quince release we will move the original repositories or the openedx-unsupported github organization to help make clear that these are no longer the default mobile application builds moving forward. A corresponding DEPR ticket will be linked to this OEP once it exists. 
+
+The new mobile repositories are maintained by Raccoon Gang and can be accessed here:
+* https://github.com/raccoongang/educationx-app-ios
+* https://github.com/raccoongang/educationx-app-android
+
+* The new repositories will be migrated to the ``openedx`` github organization under the names ``openedx-app-ios`` and ``openedx-app-android``.
+* The new repositories will be maintained in accordance with OEP-55.
+* Relevant funded contributions and projects relation to the new mobile repositories will be linked from their main project README as a way to help community members find ongoing and future efforts.
+
+
+Rationale
+*********
+
+Expanded access and opportunities for learners around the globe means meeting learners where they are, and for many people that means on their only digital device, their phone. Furthermore, enabling learners to continue to learn on their desktop or laptop computers in addition to a full featured mobile experience allows us to support full day learning workflows for those with multiple devices.
+
+Similar to the way the edx-platform has undergone a modernization process over the past few years with updated build processes, micro services, micro frontends, design system tools and more, the mobile experiences will benefit from a modern architecture, being written in the latest development language tools, and set up for increased pluggability and contribution as well.
```

Comment:
[inform/question] Related to the mention of design system tools, just wanted to cross-link to [this Paragon issue](https://github.com/openedx/paragon/issues/2470) about mobile app branding as it relates to Paragon design tokens. 

That is, there may be an opportunity to treat the Paragon design system as the "source of truth" for style properties like colors, typography, etc. across both web and mobile native platforms.

The Paragon Working Group is currently working towards migrating the Open edX web-based experiences (MFEs) to use design tokens and CSS variables. I don't think it'd be a unreasonable lift to get Paragon design tokens transformed into iOS/Android compatible output that can be consumed by mobile apps. That way, when a style property in the design system changes, its update can more readily propagate across cross-platform with less maintenance required. 

Do you think it's worth expanding on how design system tools (e.g., design tokens) could play a role here, or is that too "in the weeds" for the purpose of this OEP?

Reply:
Good question I think adding that detail might be in the weeds for the scope of the OEP but the issue in paragon that Ed created is a space we can continue the conversation toward the goal you described of a single source of truth for that kind of design meta. 

Others can chime in if they feel like additional detail should be added. Thanks for reviewing!

Reply:
Makes sense, sounds good!

Reply:
@adamstankiewicz @marcotuts @volodymyr.chekyrta Non-blocking, but even as someone who is mostly out of the loop on MFEs and the mobile apps, I have the impression that Open edX's move towards Paragon design tokens is pivotal in terms of re-establishing a consistent branding interface for the project. Confusion around theming & branding is consistently identified as one of operators' major pain points ever since we started phasing out comprehensive theming.

What do you folks think of a high-level statement in the OEP stating the design tokens (or something like it) will be supported? For example:

> In support of a consistent platform-wide branding interface, the experiences will implement branding via Paragon design tokens once tooling for the tokens reaches maturity.

or, to be more general:

> The experiences will strive to support a branding interface consistent with other Open edX frontend applications based on recommendations from the Frontend and Paragon Working Groups.

Reply:
> Confusion around theming & branding is consistently identified as one of operators' major pain points ever since we started phasing out comprehensive theming.

@kdmccormick Definitely. I think aligning on how we communicate about Paragon could help begin to mitigate this confusion. For example, perhaps rather than saying "theme-ready" and other "theming" related messaging throughout the documentation, we adopt "brand-ready" and "branding", for example, as an incremental step:

"An accessible, ~theme-ready~ brand-ready, and open source design system built for learning applications."

Reply:
> What do you folks think of a high-level statement in the OEP stating the design tokens (or something like it) will be supported?

@kdmccormick @marcotuts @volodymyr-chekyrta I would be in favor of including an explicit albeit brief, high-level mention of the design system (also, not blocking feedback though 😄). Perhaps a blend of your two suggestions, Kyle. E.g. something along the lines of:

> In support of a consistent platform-wide branding interface, the mobile experiences will strive to identify opportunities to consume and extend the Paragon design system, helping to ensure consistent and maintainable UX across channels (i.e., mobile, web).

Reply:
adding in next commit / update